config.js

```diff
@@ -381,11 +381,25 @@ config.SEND_EVENTS_REMOTESYS = true;
 config.PROMETHEUS_ENABLED = true;
 config.PROMETHEUS_PREFIX = 'NooBaa_';
 
+//////////////////////////////
+// OAUTH RELATES            //
+//////////////////////////////
+
+config.OAUTH_AUTHORIZATION_ENDPOINT = '/oauth/authorize';
```

Comment:
The current implementation is following the official documentation for OpenShift 3.11 
https://docs.openshift.com/container-platform/3.11/architecture/additional_concepts/authentication.html

Please note that the actual hostname for the endpoint is discovered by the NooBaa deploy script (src/deploy/NVA_build/noobaa_deploy_k8s.sh)
The only part that I hardcoded is the path to the token and authorize paths which are properly defined in the attached article above.

Your suggestion to use the .well-known endpoint was discussed but was rejected because it introduces a new endpoint to be discovered, the well-known endpoint, which replaces the currently discovered auth server endpoint (by querying the openshift-authentication route)

Discovering the auth server directly allows us to remove one hoop from the process (inside the NooBaa server) and have a lower failure rate.

In the future with the introduction of the NooBaa operator, we may change our minds because all of the discovery tasks will move to the operator implementation.

Reply:
> The only part that I hardcoded is the path to the token and authorize paths which are properly defined in the attached article above.

Looking at the 3.x docs does not help you with 4.x.  The products are effectively unrelated.

> Your suggestion to use the .well-known endpoint was discussed but was rejected because it introduces a new endpoint to be discovered, the well-known endpoint, which replaces the currently discovered auth server endpoint (by querying the openshift-authentication route)

I can go change the name of the route to be randomly generated per cluster if it makes the point more clear - the name is not an API.  Do not rely on it (or anything in that namespace - or even the name of the namespace itself).

> Discovering the auth server directly allows us to remove one hoop from the process (inside the NooBaa server) and have a lower failure rate.

You are making a false assumption here - the OAuth server used by the cluster can change.  It is replaceable.  In the future replacing it will actually become common for a certain set of customers.

> In the future with the introduction of the NooBaa operator, we may change our minds because all of the discovery tasks will move to the operator implementation.

It is fine if you do not want to do constant discovery but at least grab the correct endpoints from the output of `oc get --raw '/.well-known/oauth-authorization-server'`.

Reply:
@nb-ohad Can you explain what is the problem to call `oc get --raw '/.well-known/oauth-authorization-server'` when the web server starts and use that url? I'm not seeing the problem.

Reply:
@guymguym As it is now, we do not support oc calls in our script or even assume the oc tool exists on the machine that runs the script. The only tools we have dependencies on are kubectl and jq (to parse JSON data).
The alternative is to use curl but then we need to discover the address of the cluster master which will be the endpoint for such a request (I believe finding this address can be done using kubectl)

Reply:
`get` is an upstream command so `kubectl get --raw '/.well-known/oauth-authorization-server'` will also work.

Reply:
Ok, as part of the changes to split the discovery to inject env vars for the authorization and token endpoints separately I will also change the discovery method, in the deploy script, to use  ```kubectl get --raw '/.well-known/oauth-authorization-server'```